docs/website/versioned_docs/version-3.0.0/command-reference/dev.md

```diff
@@ -43,6 +43,17 @@ In the above example, three things have happened:
 You can press Ctrl-c at any time to terminate the development session. The command can take a few moment to terminate, as it
 will first delete all resources deployed into the cluster for this session before terminating.
 
+### Sustituting variables
+
+The Devfile can define variables to make the Devfile parameterizable. The Devfile can define values for these variables, and you 
+can override the values for variables from the command line when running `odo dev`, using the `--var` and `--var-file` options.
+
+The `--var` option takes a comma-separated list of `variable=value` pairs, where `=value` is optional. If the `=value` is omitted, the value is extracted from the environment variable named `variable`. In this case, if the environment variable with this name is not defined, the value defined into the Devfile will be used.
```

Comment:
Is the `--var` option really a comma-separated list of pairs?

I tried using a comma-separated list of `variable=value` pairs, but my `odo deploy` errored out with the message below, suggesting that it did not handle the comma separator:
```
❯ odo deploy --var RESOURCE_NAME=rm3l-dotnet,CONTAINER_PORT=1234
...
 ✗  failed to create service(s) associated with the component: Deployment.apps "rm3l-dotnet,CONTAINER_PORT=1234" is invalid
...

```

From the `cobra` doc, if we want to support both repeatable arguments and comma-separated values, we should use [`StringSliceVar`](https://pkg.go.dev/github.com/spf13/pflag#StringSliceVar) instead of [`StringArrayVar`](https://pkg.go.dev/github.com/spf13/pflag#StringArrayVar).
But I am not sure having a comma-separated list is the expected behavior, is it?

Also, I think providing an example of usage in this document could be helpful.

Reply:
You are right, the expected behaviour is to have only repeatable args, to be able to have commas in values. I fixed the doc and added some examples.